crates/serde/src/dependencies.rs

```diff
@@ -0,0 +1,179 @@
+//! Types for working with component dependencies.
+
+use crate::KebabId;
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use wasm_pkg_common::package::PackageRef;
+
+/// Name of an import package dependency.
+///
+/// For example: `foo:bar/baz@0.1.0`, `foo:bar/baz`, `foo:bar@0.1.0`, `foo:bar`.
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+#[serde(into = "String", try_from = "String")]
+pub struct DependencyPackageName {
```

Comment:
At the risk of opening a huge can of worms, is `Package` correct here?  It seems like part of it is the package, but it might also contain an interface.

Reply:
Short of `DependencyPackageOrInterfaceName` i don't have any better suggestions. Do you have any?

Reply:
Would `DependencyName` work?  Not a blocker either way.

Reply:
Unfortunately `DependencyName` is used below to mean either a plain name or this package form.